src/AWSMSKAuthTokenGeneratorAsync.cs

```diff
@@ -0,0 +1,225 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using Amazon;
+using Amazon.Runtime;
+using Amazon.Runtime.CredentialManagement;
+using Amazon.Runtime.Internal;
+using Amazon.Runtime.Internal.Auth;
+using Amazon.Runtime.Internal.Util;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+using System.Globalization;
+
+namespace AWS.MSK.Auth
+{
+    /// <summary>
+    /// Provides Auth tokens for IAM authentication against an MSK cluster.
+    /// </summary>
+    public class AWSMSKAuthTokenGeneratorAsync
```

Comment:
I'd recommend consolidating these two classes down to one. 

There are some shared concerns that don't need to be duplicated like:
* The constants at the top
* The `private GenerateMSKAuthTokenRequest`
* Most of the contents of ` GenerateAuthTokenFromCredentialsProvider` once we've gotten the `immutableCredentials`. This is the area with the most complexity, so I think it'd be good to consolidate for future updates in case the signing steps change.

Something like:
```
public (string, long) GenerateAuthToken(RegionEndpoint region, bool awsDebugCreds = false)
{
    // Do the STS operation with the GetAwaiter().GetResult(); workaround, since it's necessary here
    
    // Call a sync helper
    return  GenerateAuthTokenFromCredentialsProvider(() => credentials, region);
}

public async Task<(string, long)> GenerateAuthTokenAsync(RegionEndpoint region, bool awsDebugCreds = false)
{
    // await the async STS operation
    
    // Call an async helper
    return  GenerateAuthTokenFromCredentialsProviderAsync(() => credentials, region);
}

public async Task<(string, long)> GenerateAuthTokenFromCredentialsProviderAsync(Func<AWSCredentials> credentialsProvider, RegionEndpoint region)
{
     // await the GetCredentialsAsync call

    // call into the sync version with those immutable credentials, which is the actual signing logic
}
```

Reply:
We have a similar pattern for the S3 presigning in the AWS SDK for .NET in [AmazonS3Client.Extensions.cs](https://github.com/aws/aws-sdk-net/blob/06a8daa84cd9e382ae2cb89c0cfa966ebbd217aa/sdk/src/Services/S3/Custom/AmazonS3Client.Extensions.cs).

The async and sync codepaths both ultimately call into [`ReturnSigningResult`](https://github.com/aws/aws-sdk-net/blob/06a8daa84cd9e382ae2cb89c0cfa966ebbd217aa/sdk/src/Services/S3/Custom/AmazonS3Client.Extensions.cs#L371) for the signing logic once they've resolved any sync vs. async data they need.

Reply:
@ashovlin sorry for the late replay because i was off.
I followed this approach for future refactors and just to be as explicit as possible for the sync and async versions.
But i agree wit your idea, so since you are already using it people are familiar with it.
As soon as i find some time I will do the change and let you know.
Thanks

Reply:
@orges-kreka are you still working on this?

Reply:
@sankalpbhatia sorry I got lost in my daily job.
Yes i did the changes.

@ashovlin please can you take a look and let me know if you have any comment

Thanks